central/cve/image/v2/datastore/datastore_impl.go

```diff
@@ -0,0 +1,85 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/stackrox/rox/central/cve/common"
+	"github.com/stackrox/rox/central/cve/image/v2/datastore/search"
+	"github.com/stackrox/rox/central/cve/image/v2/datastore/store"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/protocompat"
+	pkgSearch "github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+type datastoreImpl struct {
+	storage  store.Store
+	searcher search.Searcher
+
+	cveSuppressionLock  sync.RWMutex
+	cveSuppressionCache common.CVESuppressionCache
+}
+
+func (ds *datastoreImpl) Search(ctx context.Context, q *v1.Query) ([]pkgSearch.Result, error) {
```

Comment:
Do you think we need any permission checks here? Since this is ported over from legacy image cve datastore which does not check permissions for ready only functions, it is fine to not add them in this PR.

Reply:
We shouldn't need them here.  We don't need any special access checks to my knowledge.  The normal ones would be handled at the store and searcher level based on the scoping resource of the tables.